src/legacy/ui/public/vis/editors/default/agg_param_react_wrapper.tsx

```diff
@@ -31,33 +31,34 @@ interface AggParamReactWrapperProps<T> {
   isInvalid: boolean;
   paramEditor: React.FunctionComponent<AggParamEditorProps<T>>;
   value: T;
-  onChange(value: T): void;
+  onChange(value?: T): void;
   setTouched(): void;
   setValidity(isValid: boolean): void;
 }
 
 function AggParamReactWrapper<T>(props: AggParamReactWrapperProps<T>) {
-  const {
-    agg,
-    aggParam,
-    indexedFields,
-    isInvalid,
-    paramEditor: ParamEditor,
-    value,
-    onChange,
-    setValidity,
-    setTouched,
-  } = props;
+  const { agg, aggParam, paramEditor: ParamEditor, onChange, setValidity, ...rest } = props;
+
+  useEffect(
+    () => {
+      if (aggParam.shouldShow && !aggParam.shouldShow(agg)) {
+        setValidity(true);
+      }
+    },
+    [agg, agg.params.field]
```

Comment:
useEffect body will be invoked if `agg` or `agg.params.field` changed. We don't need to run this code every time when each param changed because existing `shouldShow` functions are based on field value only.